### DISCUSSION TITLE
JSON Schema, Types, Snippets

### DISCUSSION THREAD

**Discussion by Klustre** [KraftyFX/aex#67] [Ideas]
Rather than serializing objects to AEX I would want to generate AEX objects programmatically. To make sure they're valid I think it would be great to validate them againts a JSON schema and to have types for AEX objects. Possibly also snippets for all the items.

Related: do the objects require AEX IDs? Or would AEX add these when missing?

**Comment by rafikhan** [comment]
I captured the need for schema validation and getting type safety in #15.

As for AEX ID's I looked at the source and it appears it has to be defined. This makes sense in the case when you need build out a graph of comps with nested pre-comps, etc.

**Comment by rafikhan** [comment]
@Klustre what did you mean by "snippets" exactly?

**Reply by Klustre** [reply]
VSCode snippets for all of the AE objects you can create, such as comps. This would give you the AEX object in VSCode with all the supported keys and defaults.

**Reply by Klustre** [reply]
For example writing `aexcomp` or whatever would insert:
```json
{
  "type": "aex:item:av:comp",
  "name": "Comp",
  "aexid": "comp:1",
  "duration": 4,
  "frameRate": 60,
  "height": 720,
  "width": 1280,
  "layers": []
}
```
